api/app/src/main/kotlin/packit/service/UserService.kt

```diff
@@ -110,6 +117,20 @@ class BaseUserService(
         return userRepository.saveAll(users)
     }
 
+    override fun updatePassword(username: String, updatePassword: UpdatePassword)
+    {
+        val user = userRepository.findByUsername(username)
+            ?: throw PackitException("userNotFound", HttpStatus.NOT_FOUND)
+
+        if (!passwordEncoder.matches(updatePassword.currentPassword, user.password))
+        {
+            throw PackitException("invalidPassword", HttpStatus.BAD_REQUEST)
```

Comment:
Feels like this should maybe be a 401 since it's a wrong password..? But it is embedded the the update password request body, rather than like a normal login, so maybe a 400 is ok..

Also - I'm not sure we should re-use this for a normally logged in user choosing to update their password, if that was the idea. In that case, they would already be logged in and should hit an endpoint secured with a token, rather than posting their password in the body again.

Reply:
yeah i initially had in user controller, but then realised they don't need auth.. technically yeah you are right but fundamentally we can resuse this same one... as they need they need their email and password  (same as login)... unless we make another update one that only requires new password

Reply:
I guess it's good practice to ask for old password again when they want to change it!